api/v1alpha2/osartifact_types.go

```diff
@@ -24,8 +24,17 @@ import (
 
 // OSArtifactSpec defines the desired state of OSArtifact
 type OSArtifactSpec struct {
+	// There are 3 ways to specify a Kairos image:
+
+	// Points to a prepared kairos image (e.g. a released one)
 	ImageName string `json:"imageName,omitempty"`
 
+	// Points to a vanilla (non-Kairos) image. osbuilder will try to convert this to a Kairos image
+	BaseImageName string `json:"baseImageName,omitempty"`
+
+	// Points to a Secret that contains a Dockerfile. osbuilder will build the image using that Dockerfile and will try to create a Kairos image from it.
+	BaseImageDockerfile *SecretKeySelector `json:"baseImageDockerfile,omitempty"`
```

Comment:
I'm a bit confused by this, why does it need to be a secret?

Reply:
This is the user's dockerfile, since we don't know what it contains and given ConfigMaps and Secrets have very little differences, I preferred a secret over a ConfigMap.

Reply:
gotcha, ok makes sense in case there's any confidential data from the user there